src/layout/register/register-form/RegisterForm.tsx

```diff
@@ -1,30 +1,67 @@
-import { Form } from 'react-router-dom';
-import { useForm } from 'react-hook-form';
-import './register-form.scss';
+import { FieldErrors, useForm } from 'react-hook-form';
 import * as yup from 'yup';
 import MainButton from '../../../components/buttons/MainButton';
 import { registerFormSchema } from '../../../models/schemas.yup';
 import { yupResolver } from '@hookform/resolvers/yup';
-import classnames from 'classnames';
+import './register-form.scss';
+import { useRegisterUserMutation } from '../../../api/auth/authApi';
+import { FC, useEffect, useState } from 'react';
+import { useNavigate } from 'react-router-dom';
 
-type FormData = yup.InferType<typeof registerFormSchema>;
+const USER_EXIST_ERROR_MESSAGE = 'Użytkownik o podanej nazwie już istnieje.';
+export type RegisterSchemaType = yup.InferType<typeof registerFormSchema>;
 
-const RegisterForm = () => {
+const useClearUserExistErrorEffect = (
```

Comment:
Może zamiast nazwy useClearUserExistErrorEffect to handleUserExistsError?

Reply:
- `use` musi być w nazwie ponieważ jest to `useEffect`
- to nie jest do końca "handler", to ma za zadanie wyczyścić stan `isUserExistError` gdy taki właśnie error się pojawi, bo jak się pojawi i wpisze się przykładowo złe dane, pojawiają się dwa błędy obok siebie, w tym `isUserExistError` który w takim momencie jest nieaktualny, i `useClearUserExistErrorEffect` to naprawia

Reply:
Wyślę ci filmik dokładnie z tą sytuacją